cpp/src/arrow/json/from_string.cc

```diff
@@ -1052,28 +1047,27 @@ Status ScalarFromJSONString(const std::shared_ptr<DataType>& type,
   RETURN_NOT_OK(converter->AppendValue(json_doc));
   RETURN_NOT_OK(converter->Finish(&array));
   DCHECK_EQ(array->length(), 1);
-  return array->GetScalar(0).Value(out);
+  ARROW_ASSIGN_OR_RAISE(auto out, array->GetScalar(0));
+  return out;
```

Comment:
Can we simplify this?

```suggestion
  return array->GetScalar(0);
```

Reply:
Interesting. I didn't realize you could do this but it makes sense we'd provide an automatic conversion.